app/services/products_renderer.rb

```diff
@@ -54,7 +54,7 @@ def enterprise_fee_calculator
   def filter_and_paginate(query)
     results = query.ransack(args[:q]).result
 
-    _pagy, paginated_results = pagy(
+    _pagy, paginated_results = pagy_arel(
```

Comment:
I'm wondering what this method is needed for? It looks like the old relation used just as much arel as the new one.

Reply:
Pagy wasn't correctly handling the query here, leading to some failing specs due to pagination errors. Something about the join being added was confusing it. This `pagy_arel` method just makes pagy use some slightly more nuanced query parsing, which in this case meant it could handle the join correctly.